search-enterprise/latest/upgrade/index.md

```diff
@@ -43,3 +43,69 @@ Your current Content Services stack can continue to run while you are indexing t
+
+## Zero downtime upgrade
+
+You can upgrade from Search Services 2.x without experiencing any downtime, to Search Enterprise 3.1.1 when you are using Content Services 7.2 and above.
+
+1. Start an Elasticsearch 3.1.1 instance, for more see [Overview]({% link search-enterprise/latest/install/index.md %}).
+
+    Currently your installation is using solr.
+
+    ![add-empty]({% link search-enterprise/images/add-empty-elasticsearch.png %})
+
+2. Start a mirrored environment by replicating the content respository and Content Services.
+
+    You create a mirrored environment because the upgrade will not impact the primary environment. Use the Elasticsearch instance you created as the content repository for the mirrored environment. Once you have mirrored the environment do not change the content repository and only use it in read-only mode. If you do not need to preserve the content repository then you only need to mirror Content Services.
+
+    ![mirror-acs]({% link search-enterprise/images/mirror-acs-environment.png %})
+
+3. Create an Elasticsearch index by executing a search query on the mirrored environment.
+
+    Verify the index is created and its metadata correctly reflects your data model.
+
+    > **Note:** The index is not created when you mirrored the content repository and Content Services.
+
+4. Populate the index with existing data.
+
+    The index is populated and is based on the replicated database and is achieved by starting re-indexing on the mirrored environment. For more see [re-indexing]({% link search-enterprise/latest/config/index.md %}#alfresco-re-indexing-app).
+
+    **Note:** A window displays that states the primary database does not reflect the up to date index.
+
+    ![intial-reindexing]({% link search-enterprise/images/initial-re-indexing.png %})
+
+5. Shutdown the original environment.
+
+    You are left with an Elasticsearch server with a populated index.
+
+    **Note:** The index is not yet in sync with the primary environment.
+
+    ![shutdown-mirrored]({% link search-enterprise/images/shutdown-mirrored.png %})
+
+6. Keep your index up to date with changes made through Content Services.
+
+    To do this start live indexing on the primary environment. For more see [live-indexing]({% link search-enterprise/latest/config/index.md %}#alfresco-live-indexing-app).
+
+    > **Note:** Even after starting live indexing there is still a gap from when you took a snapshot to when you started live indexing.
+
+    ![start-live-indexing]({% link search-enterprise/images/start-live-indexing.png %})
+
+7. Start re-indexing on the solr environment.
+
+    To close the gap in the Elasticsearch index start re-indexing on the solr environment. **Note:** Live indexing keeps the Elasticsearch environment index up to date.
```

Comment:
Fix the formatting for all notes so they're similar to line 89 (there are several instances).